### PR TITLE
[DependsOn]: input event & lazy refs

### PR DESCRIPTION
**Pull request by ala-n**
References now observing input event.
"Lazy" references introduced to ignore rapid input event.

### CODE REVIEW

File: aem-authoring-toolkit-assets/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/README.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ For referenced field:
 
 * `data-dependsonref` - to mark a field, that is referenced from the Query.
 * `data-dependsonreftype` - (optional) to define expected type of reference value. 
+* `data-dependsonreflazy` - (marker) attribute to mark reference as lazy. DependsOn doesn't observe rapid events like `input`.
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-assets/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/README.md:45]: (cosmetic) maybe rephrase the lase sentence as "In this case, DependsOn will not observe rapid events like `input`."

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/assets/dependson/DependsOnRef.java [hunk 2]
```diff
@@ -26,4 +26,10 @@
      * @return One of the predefined {@link DependsOnRefTypes} constants
      */
     DependsOnRefTypes type() default DependsOnRefTypes.AUTO;
+
+    /**
+     * Defines the 'dependsInRefLazy' attribute
```
Comment (smiakchilo) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/assets/dependson/DependsOnRef.java:31]: 'dependsOnRefLazy', probably)
